include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -233,6 +233,11 @@ __internal::sycl_iterator<access_mode::discard_read_write, T, Allocator> end(syc
     return __internal::sycl_iterator<access_mode::discard_read_write, T, Allocator>{buf,
                                                                                     __dpl_sycl::__get_buffer_size(buf)};
 }
+
+template <typename ValueType>
+constexpr static bool usm_allocated_vector_iterators_supported_v =
+    oneapi::dpl::__internal::__vector_iter_distinguishes_by_allocator<ValueType*>::value;
```

Comment:
BTW, what if `ValueType` is a pointer to the host data?
It is rather a kind of general question connected to "Pass Data to algorithms" topic.

Reply:
If I understand correctly, you suggest the following type:

`std::vector<int*, usm_allocator<int, ...>>`

In theory, a vector of host pointers will be very similar to a vector of integers I think.  

In principle, as long as they are not dereferencing the pointers on the device, there is nothing wrong with this.